### PR TITLE
Document new operators

### PR DESCRIPTION
**Pull request by pontusmelke**
Add docs for the new operators, bfs expand and union node by labels scan.

### CODE REVIEW

**Comment by pontusmelke** [comment]
[Document UnionNodeByLabelsScan](https://trello.com/c/iXnVZ2HT/3537-document-unionnodebylabelsscan)

**Comment by pontusmelke** [comment]
[Document BFSVarExpand](https://trello.com/c/iQgas1lp/3538-document-bfsvarexpand)

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 1]
```diff
@@ -578,6 +578,19 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
...
   }
 
+  @Test def unionNodeByLabelScan() {
+    executePreparationQueries {
+      List("CREATE LOOKUP INDEX lookup_index_name FOR (n) ON EACH labels(n)")
+    }
+
+    profileQuery(
+      title = "Union Node By Labels Scan",
+      text = """The `UnionNodeByLabelsScan` operator fetches all nodes having at least one of the provided labels from the node label index.""".stripMargin,
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]:
```suggestion
      text = """The `UnionNodeByLabelsScan` operator fetches all nodes that have at least one of the provided labels from the node label index.""".stripMargin,
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 2]
```diff
@@ -1084,6 +1097,26 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     )
   }
 
+  @Test def bfsVarlengthExpandPruning() {
+    profileQuery(
+      title = "Breadth First VarLength Expand Pruning",
+      text =
+        """Given a start node, the `VarLengthExpand(Pruning, BFS)` operator will traverse variable-length relationships much like the <<query-plan-varlength-expand-all, `VarLengthExpand(All)`>> operator.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]:
```suggestion
        """Given a start node, the `VarLengthExpand(Pruning, BFS)` operator traverses variable-length relationships much like the <<query-plan-varlength-expand-all, `VarLengthExpand(All)`>> operator.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 3]
```diff
@@ -1084,6 +1097,26 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
...
+          |However, as an optimization, it will instead perform a breadth-first search (BFS) and while expanding some paths will not be explored if they are guaranteed to produce an end node that has already been found (by means of a previous path traversal).
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]:
```suggestion
          |However, as an optimization, it instead performs a breadth-first search (BFS) and while expanding, some paths are not explored if they are guaranteed to produce an end node that has already been found (by means of a previous path traversal).
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 4]
```diff
@@ -1084,6 +1097,26 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
...
+          |This will only be used in cases where the individual paths are not of interest.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]:
```suggestion
          |This is only used in cases where the individual paths are not of interest.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 5]
```diff
@@ -1084,6 +1097,26 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
...
+          |
+          |This kind of expand is only planned when:
+          |
+          |* The individual paths are not of interest.
+          |* The relationships have an upper bound.
+          |* The lower bound is either 0 or 1 (default).
+          |
+          |This operator guarantees that all the end nodes produced will be unique.""".stripMargin,
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]:
```suggestion
          |This operator guarantees that all the end nodes produced are unique.""".stripMargin,
```

**Review by AlexicaWright — COMMENTED** [review]
Just some editorial comments for clarity and grammar.

**Pull request merged**